File: web/src/api/sportCity.js
import service from '@/utils/request'

// @Tags SportCity
// @Summary 创建SportCity
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SportCity true "创建SportCity"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /sportCity/createSportCity [post]
export const createSportCity = (data) => {
  return service({
    url: '/sportCity/createSportCity',
    method: 'post',
    data
  })
}

// @Tags SportCity
// @Summary 删除SportCity
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SportCity true "删除SportCity"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /sportCity/deleteSportCity [delete]
export const deleteSportCity = (data) => {
  return service({
    url: '/sportCity/deleteSportCity',
    method: 'delete',
    data
  })
}

// @Tags SportCity
// @Summary 删除SportCity
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除SportCity"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /sportCity/deleteSportCity [delete]
export const deleteSportCityByIds = (data) => {
  return service({
    url: '/sportCity/deleteSportCityByIds',
    method: 'delete',
    data
  })
}

// @Tags SportCity
// @Summary 更新SportCity
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SportCity true "更新SportCity"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /sportCity/updateSportCity [put]
export const updateSportCity = (data) => {
  return service({
    url: '/sportCity/updateSportCity',
    method: 'put',
    data
  })
}

// @Tags SportCity
// @Summary 用id查询SportCity
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SportCity true "用id查询SportCity"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /sportCity/findSportCity [get]
export const findSportCity = (params) => {
  return service({
    url: '/sportCity/findSportCity',
    method: 'get',
    params
  })
}

// @Tags SportCity
// @Summary 分页获取SportCity列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取SportCity列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /sportCity/getSportCityList [get]
export const getSportCityList = (params) => {
  return service({
    url: '/sportCity/getSportCityList',
    method: 'get',
    params
  })
}

File: server/router/sport_city.go
package router

import (
	"gin-vue-admin/api/v1"
	"gin-vue-admin/middleware"
	"github.com/gin-gonic/gin"
)

// InitSportCityRouter 初始化 SportCity 路由信息
func InitSportCityRouter(Router *gin.RouterGroup) {
	SportCityRouter := Router.Group("sportCity").Use(middleware.OperationRecord())
	{
		SportCityRouter.POST("createSportCity", v1.CreateSportCity)   // 新建SportCity
		SportCityRouter.DELETE("deleteSportCity", v1.DeleteSportCity) // 删除SportCity
		SportCityRouter.DELETE("deleteSportCityByIds", v1.DeleteSportCityByIds) // 批量删除SportCity
		SportCityRouter.PUT("updateSportCity", v1.UpdateSportCity)    // 更新SportCity
		SportCityRouter.GET("findSportCity", v1.FindSportCity)        // 根据ID获取SportCity
		SportCityRouter.GET("getSportCityList", v1.GetSportCityList)  // 获取SportCity列表
	}
}
